client/plots/matrix/matrix.legend.js

```diff
@@ -143,6 +143,54 @@ export function getLegendData(legendGroups, refs, self) {
 		const keys = Object.keys(legend.values).sort()
 		const hasScale = Object.values(legend.values).find(v => v.scale)
 		if (hasScale) {
+			if (legend === legendGroups?.CNV) {
+				/** Combine gain and loss here to avoid rendering issues */
+				const legendValues = Object.values(legend.values)
+				const findLegend = type =>
+					legendValues.find(v => v.label.toLowerCase().includes(type) || v.key.toLowerCase().includes(type))
+
+				const gain = findLegend('gain')
+				const loss = findLegend('loss')
+
+				if (gain?.scale && loss?.scale) {
+					const gainColors = getColors(gain)
+					const lossColors = getColors(loss)
+					const colors = [lossColors[1], 'white', gainColors[1]]
+					const domain = setColorScaleDomain(loss.maxLabel, gain.maxLabel, [0, 0], colors)
```

Comment:
should use `loss.domain[1]` and `gain.domain[1]`, the actual numeric values instead of extracting presumed numeric values from string labels

Reply:
@siosonel, the problem with using the domain for the cnv values is they are often not the real data. Usually the domains were set to [0,1]. Only the min and max label contained the real data.

Reply:
Based on `matrix.cell.js` (see screenshot below), `loss.maxLabel` is equal to `loss.domain[1]` or `-minLoss`, and `gain.maxLabel === gain.domain[1]` or `maxGain`, and these refer the actual numeric values. Whereas variable names with `"label"` in it may eventually include units or comma-separators, these are not guaranteed to be fully numeric values in the future. And by using numbers that are taken directly from the `domain[1]`, which was used for the scale to color matrix CNV cells, it ensures that the legend colors match the rendered data value colors. But if there are other technical concerns about using domain, we can discuss.

![Screenshot 2024-12-03 at 8 33 22 PM](https://github.com/user-attachments/assets/d577550f-e227-45ca-a1aa-8badd8e3840d)
 `setColorScaleDomain()`.

Reply:
That's still not the real data in every instance. `-minloss` converts the real minimum CNV loss value to a positive value in every instance. The maxLabel is the correct domain[1] for each item.

Reply:
I intentionally left the rendering code alone since how the scales are used are not the same.

Reply:
addressed in my fix